src/eda_server/api/activation.py

```diff
@@ -35,6 +68,8 @@ async def create_activation(
     activation: schema.ActivationCreate,
     db: AsyncSession = Depends(get_db_session),
 ):
+    await dependent_object_exists_or_exception(db, activation)
```

Comment:
Since this function performs 4 SQL queries, it may be reasonable to call it in response of `IntegrityError` exception to clarify the exception reason.